### ISSUE TITLE
TESTERS NEEDED. Anyone with UART hooked up to BK7231? BK7231T or N, or both? New flash tool

### ISSUE THREAD

**Issue by openshwprojects** [openshwprojects/OpenBK7231T_App#579]
I am releasing a new flash tool now, can anyone help with testing?
https://github.com/openshwprojects/BK7231GUIFlashTool
See readme on that repository.

**Comment by HumbleDeer** [comment]
Works fine.

**Comment by ByteMangler** [comment]
tried it today with a CB3S module BK7231N. Worked as advertised.

**Comment by Perfunctory85** [comment]
tried it today with a CBU module BK7231N. Doesnt work....  the tool says "failed to set baudrate". Ive tried all baudrate options. anyone can help? Tried two different usb->uart sticks

**Comment by ByteMangler** [comment]
I found that the CP2102 usb->serial device I use doesn't like 9216k so I set the baud rate to 115200. 
@Perfunctory85, are you sure your rx/tx wiring is correct? Very easy to get it the wrong way around! If in doubt, swap the wires.

**Comment by Perfunctory85** [comment]
> I found that the CP2102 usb->serial device I use doesn't like 9216k so I set the baud rate to 115200. @Perfunctory85, are you sure your rx/tx wiring is correct? Very easy to get it the wrong way around! If in doubt, swap the wires.

Ive tried every baudrate :( And of course, i swap the wires and check it twice... Doesnt work :(

**Comment by Perfunctory85** [comment]
![IMG20230106215533](https://user-images.githubusercontent.com/105150364/211098624-c7d299d2-26c6-4c24-b81e-2192febf8eb7.jpg)
![IMG20230106215538](https://user-images.githubusercontent.com/105150364/211098630-b5d4f4a5-dcbb-4820-b3c4-c7c6b04db709.jpg)

**Comment by ByteMangler** [comment]
@Perfunctory85, the jumper on your usb->serial board is set to 5V. Not sure how the BK7231 reacts to over-voltage on the serial pins. 
How do you reset the BK7231? cycle power or gnd CEN? 
I gnd CEN and sometimes it takes a few tries.

**Comment by Perfunctory85** [comment]
Yes, you are right. Now i put it on 3.3V. But on this way, i cant get the bus.  Tried CEN -> GND and Powercyle.

output auf UART is about 3 V. Also if the jumper is connected to 5v

**Comment by ByteMangler** [comment]
maybe your device is also using the serial pins and the usb->serial is fighting with it? If that is the case, then it may be necessary to cut a track or desolder the module. Neither option is ideal.

An oscilloscope would be handy to check if there is any fighting.

**Comment by Perfunctory85** [comment]
Thats it. Ive desolder the modul an now i can flash it. Thanks. I hope i get it resoldered :D

**Comment by ByteMangler** [comment]
@Perfunctory85  good to hear you were successful.

@openshwprojects might be worth noting my tests were done using an iMac with Windows7 64bit running in a virtual machine with Parallels as the hypervisor and a CP2102 usb->serial chip. As previously mentioned, I needed to set the baudrate to 115200.

**Comment by dvcomp** [comment]
Hi!

Not working for me. bk7231tools worked with the same environment. WBLC5 module.

**Comment by crunchtime2k** [comment]
Works fantastic on CBU module, thanks

**Comment by roobbb** [comment]
Hello.

I tried to follow this instruction: https://www.elektroda.com/rtvforum/topic3912748.html#20145042
Creating a backup with BK7231Flasher.exe works fine. But it creates a file which has 2.048KB. BKwriter 1.60 even dumps fine a file with 1.192KB. I thought it should be the same file size in any case.

Flashing unfortunately always fails for me, no matter what baud rate I use:
115200
```
All selected sectors erased!
CheckRespond_FlashWrite4K: bad value returned?
Writing sector 0x00... Writing sector 0 failed!
Writing file data to chip failed.
```
or 921600
```
Writing sector 0x00... ok! CheckRespond_FlashWrite4K: bad value returned?
Writing sector 0x1000... Writing sector 4096 failed!
Writing file data to chip failed.
```
or 1500000
```
Writing sector 0x00... ok! Writing sector 0x1000... ok! CheckRespond_FlashWrite4K: bad value returned?
Writing sector 0x2000... Writing sector 8192 failed!
Writing file data to chip failed.
```
I have no idea what I am missing or doing wrong.
It seems flashing via python or bk_writer1.60.exe even doesn't work. So it might not be a problem of BK7231Flasher.exe, but I thought I should report here the above behavior. 

Kind regards
roobbb

----
![flashing_bk7231n_fails](https://user-images.githubusercontent.com/43150729/224509352-463f6ac9-33c0-4fe4-8a1f-e25d5ccd5e75.jpg)

**Comment by roobbb** [comment]
OK, I think I have found my mistake: As stated in the 1st post of the forum's thread above, I had no Vcc connected. Reading flash worked fine that way. Now, with Vcc connected to 3.3V flashing works fine too with this tool 😅

regards
roobbb
